pkg/aws/cloud.go

```diff
@@ -42,24 +43,37 @@ type Cloud interface {
 
 // NewCloud constructs new Cloud implementation.
 func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud, error) {
-	metadataSess := session.Must(session.NewSession(aws.NewConfig()))
-	metadata := services.NewEC2Metadata(metadataSess)
-	if len(cfg.Region) == 0 {
-		region, err := metadata.Region()
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to introspect region from EC2Metadata, specify --aws-region instead if EC2Metadata is unavailable")
-		}
-		cfg.Region = region
-	}
-
+	metadata := (services.EC2Metadata)(nil)
 	if len(cfg.VpcID) == 0 {
+		metadataSess := session.Must(session.NewSession(aws.NewConfig()))
```

Comment:
This code path is very likely, so  I don't see any benefit of moving these two variables inside the if condition.